hazelcast/test/src/HazelcastTests7.cpp

```diff
@@ -610,6 +639,34 @@ TEST_F(ClientQueueTest, testListener)
     ASSERT_NO_THROW(q->destroy().get());
 }
 
+TEST_F(ClientQueueTest, testListenerOnRemoved)
+{
+    ASSERT_EQ(0, q->size().get());
+    constexpr int num_of_entry = 5;
+    boost::latch latch1(num_of_entry);
+
+    auto listener = item_listener().on_removed(
+      [&latch1](item_event&& item_event) { latch1.count_down(); });
+
+    for (int i = 0; i < num_of_entry; i++) {
+        ASSERT_TRUE(
+          q->offer(std::string("event_item") + std::to_string(i)).get());
+    }
+
+    auto id = q->add_item_listener(std::move(listener), true).get();
+
+    for (int i = 0; i < num_of_entry; i++) {
+        ASSERT_TRUE(
+          q->remove(std::string("event_item") + std::to_string(i)).get());
+    }
+
+    ASSERT_OPEN_EVENTUALLY(latch1);
+    ASSERT_TRUE(q->remove_item_listener(id).get());
+
+    // added for test coverage
+    ASSERT_NO_THROW(q->destroy().get());
```

Comment:
This `destroy` is `ClientProxy::destroy` but the uncovered one is `TransactionalObject::destroy()`. So this check can be removed.
```suggestion
```

Reply:
Fixed.

Reply:
It is there, r u it is fixed ?

Reply:
You are too quick :). I have pushed now.